class.jetpack.php

```diff
@@ -4104,6 +4105,14 @@ function build_connect_url( $raw = false, $redirect = false, $from = false, $reg
 
 		if ( $register || ! $token || ! $site_id ) {
 			$url = Jetpack::nonce_url_no_esc( Jetpack::admin_url( 'action=register' ), 'jetpack-register' );
+
+			if ( ! empty( $redirect ) ) {
+				$validated_redirect = wp_validate_redirect( esc_url_raw( $redirect ), false );
+				$url = ! empty( $redirect )
```

Comment:
@zinigor yes, thanks! I updated things in 9b853a2d585237b40df3f953371fd3f607e2ae76